src/TwigComponent/doc/index.rst

```diff
@@ -337,6 +337,19 @@ You can even give the block default content. See
 :ref:`Passing HTML to Components via Block <embedded-components>`
 for more info.
 
+Composition
+~~~~~~~~~~~
+
+One of the most interesting things to do when working with components is composition.
+
+.. code-block:: html+twig
+
+    <twig:Alert>
+        <twig:Icon name="success"/> Nice work!
+    </twig:Alert>
+
+You can have components inside your components! This is great! However, be aware that every component lives in its own context. This means that your child component does not know that it is inside a parent component; you cannot use any method like parent() inside the child component to access the parent. You should keep your components independent of the context. This rule is really important to respect, as it will guarantee that your components will always work the same way regardless of where you use them. If you need to access some parent data, you should pass properties down to your child or use events.
```

Comment:
Nice ! 

Two small comments:
- maybe split in two lines, to emphasis some words, and improve even more readability ? 
- i think we want to add a bit of nuance here or you may frighten people 😆 

Instead of 
> You should keep your components independent of the context 

Maybe 

> You should try to keep (...)

or  

> It is generally recommanded to ...

Same for the last sentence ?

Reply:
It's more like `you must`. Actually using the context is just unsupported. And I slightly disagree with the tone of this piece of documentation that suggest composition to explain inheritance does not work.

IMO it should be named something like `Composition enforced` if you want to get people straight to your point 😉 .

Reply:
In fact it's a bit more complicated than that.

There is a difference between _anonymous_ and _class-based_ components, there is a difference betwen "`include`" and "`embed`" components, and i don't even start with `live` ones haha

But this works perfectly

```twig

    {% set aa = 'AA' %}

    <twig:Browser>

        {{ dump(aa) }}

    </twig:Browser>

```

<img width="373" alt="Capture d’écran 2024-06-09 à 14 18 56" src="https://github.com/symfony/ux/assets/1359581/757357bc-8c51-4ccd-b4aa-3283462aa5dd">


A component should not know an outside one existence.

There it's something else :)

Reply:
Ok, it just feels weird to me to have an example unrelated to the comment following then.

Reply:
My point was only that I prefer to avoid binary formulations, as things in life are rarely that simple, and few things in those components are that clear-cut. :)